src/config/context/model.ts

```diff
@@ -0,0 +1,58 @@
+import { injectable, registry, } from 'tsyringe';
+import fs from 'fs';
+import { CONTEXTFILE_PATH } from '../../constants';
+
+export interface IContext {
+  readonly current?: string,
+  readonly store: {
+    [name: string]: string
+  }
+}
+
+export class Context implements IContext {
+  current?: string | undefined;
+  store: { [name: string]: string; };
+
+  constructor(ctx: IContext) {
+    this.current = ctx.current;
+    this.store = ctx.store;
+  }
+
+  getContext(contextName: string) {
+    return this.store[contextName as string];
+  }
+}
+
+export interface IContextAllocator {
+  contextFilePath?: string
+  load: () => Context | undefined
+  save: (context: Context) => boolean
+}
+
+@injectable()
+@registry([
+  {
+    token: 'IContextAllocator',
+    useToken: ContextAllocator
```

Comment:
Also, you wrote in the PR description:

> To support the future ambition of loading context files from the root directory and to be sharable I have created a separate class ContextAllocator to load context files accordingly. For now, it is loading the context file from the root file itself.

so in the future, will the `IContextAllocator` interface be implemented by other classes that can also be injected? I don't see the functionality in the `tsyringe` to have multiple implementations for a single token. Really sorry, but I don't understand this sentence:

> To support the future ambition of loading context files from the root directory and to be sharable I have created a separate class ContextAllocator to load context files accordingly

to accomplish goal by `IContextAllocator` token.

Reply:
So what I was thinking is that there are two cases where the context is used, one is to lookup context for loading input (only read functions) and the other is in the context commands where the context store is manipulated. So `ContextAllocator.load` is returning the `context` class which allows only to read data. And creating instance of the `ContextService` allows to read and manipulate the context. 

> To support the future ambition of loading context files from the root directory and to be sharable I have created a separate class ContextAllocator to load context files accordingly. For now, it is loading the context file from the root file itself.

The idea is still hazy but, this is something like having a config file, somewhat like what `jest.config.js` has where the user creates a `.asyncapi` file in the working directory writing all context. And this file could also be read over the internet. I was thinking that making any change in logic for locating the context file would be easy if implemented like this.

Reply:
As I understand it, the implementation currently allows to read single configs and will be further developed to e.g. support operation on two contexts? Am I corect?

Here comes my question. Why should you treat it as an interface injection (by `IContextAllocator` token), and not as a simple `ContextAllocator` token? Interface injection is always used when there are several implementations for this interface, which we want to inject in appropriate places in the application. Could you describe this decision. Thanks!

Reply:
I am new to the DI concepts and I don't know if this is the right way to do this. I injected the `IContextAllocator` token because I wanted to pass in mock `ContextAllocator` during tests. when I was passing as a Simple ContextAllocator token it was not working. 

So I did something like this 

```
class MockPositiveContextAllocator implements IContextAllocator {
  load() {
    return new Context({
      current: 'test',
      store: {
        test: './test/specification.yml',
        check: './test/specfication.yml'
      }
    });
  }

  save(context: Context) {
    return context instanceof Context;
  }
}

```

Reply:
Ok I see

> I am new to the DI concepts and I don't know if this is the right way to do this.

No worries, good thing we explained to each other :) As I see in the code, you can still make this:

```ts
@injectable()
export class ContextService {
  private _context: Context | undefined
  constructor(
    @inject(ContextAllocator) private contextAllocator: IContextAllocator
  ) {
    this._context = this.contextAllocator.load();
  }
  
  ...
}
```

and then remove `@registry` decorator above `ContextAllocator` class. In this approach you can still use the `MockPositiveContextAllocator` mock in the test with

```tx
ctxService = new ContextService(new MockPositiveContextAllocator());
```

:) I don't know why it wasn't work for you in previous implementation, but I didn't see it, so I cannot say why 😅 

Please try this approach :)